site/content/en/docs/concepts/resource_flavor.md

```diff
@@ -3,25 +3,44 @@ title: "Resource Flavor"
-an [empty ResourceFlavor](#empty-resourceflavor) instead of adding labels to custom ResourceFlavors.
+If the resources in your cluster are homogeneous, you can use an [empty ResourceFlavor](#empty-resourceflavor) instead of adding labels to custom ResourceFlavors.
 {{% /alert %}}
 
-A sample ResourceFlavor looks like the following:
+## ResourceFlavor tolerations for automatic scheduling
+
+**Requires Kubernetes 1.23 or newer**
+
+This approach may be best for teams that wish to schedule pods onto the appropriate nodes automatically. However, a limitation arises when multiple types of specialized hardware are present, such as two different nvidia.com/gpu resources present in the cluster, i.e., T4 and A100 GPUs. The system may not differentiate between them, meaning the pods could be scheduled on any of both types of hardware.
+
+To associate a ResourceFlavor with a subset of nodes of your cluster, you can configure the `.spec.nodeLabels` field with matching node labels that uniquely identify the nodes. If you are using [cluster autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler) (or equivalent controllers), make sure that the controller is configured to add those labels when adding new nodes.
+
+To guarantee that the Pods in the [Workload](/docs/concepts/workload) run on the nodes associated to the flavor that Kueue selected, Kueue performs the following steps:
+
+1. When admitting a Workload, Kueue evaluates the [`.nodeSelector`](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector) and [`.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution`](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity) fields in the PodSpecs of your [Workload](/docs/concepts/workload) against the ResourceFlavor labels. `ResourceFlavors` that don't match the node affinity of the Workload cannot be assigned to a Workload's podSet.
+
+
+2. Once the Workload is admitted:
+   - Kueue adds the ResourceFlavor labels to the `.nodeSelector` of the underlying Workload Pod templates. This occurs if the Workload didn't specify the `ResourceFlavor` labels already as part of its nodeSelector.
+
+     For example, for a [batch/v1.Job](https://kubernetes.io/docs/concepts/workloads/controllers/job/), Kueue adds the labels to the `.spec.template.spec.nodeSelector` field. This guarantees that the workload's Pods can only be scheduled on the nodes targeted by the flavor that Kueue assigned to the Workload.
+
+   - Kueue adds the tolerations to the underlying Workload Pod templates.
+
+     For example, for a [batch/v1.Job](https://kubernetes.io/docs/concepts/workloads/controllers/job/), Kueue adds the tolerations to the `.spec.template.spec.tolerations` field. This allows that the workloads Pods to be scheduled on nodes having specific taints.
+
+A sample ResourceFlavor of this type looks like the following:
 
 ```yaml
 apiVersion: kueue.x-k8s.io/v1beta1
```

Comment:
Should we move this to a samples and use that here? That follows our code in workloads.

Reply:
Do you mean like this for example?
https://github.com/kubernetes-sigs/kueue/blob/92e194db4f47e54bddd05149175ef9052ac96c2d/site/content/en/docs/tasks/manage/monitor_pending_workloads/pending_workloads_in_status.md?plain=1#L38-L42 

My only concern with this, is that then the `yaml` and the fields are then not displayed in the docs. I could add both, the `yaml` and the file that can be applied. Does that sound good?

Reply:
Something like: https://github.com/kubernetes-sigs/kueue/blob/main/site/content/en/docs/tasks/run/jobsets.md

Where we define the sample yaml in the static folder and import it in the docs. The website renders the YAML but we also have examples users can apply without reading the website.

Reply:
Ohh I understand. Hmm, for this to be a "working" ResourceFlavor, we would need to know which `nodeLabels` to use in the YAML. `nodeLabels` being used to associate the ResourceFlavor with the node in the cluster. A generic one wouldn't work, so the user can't just apply the file and see real results. - The only way I see, is to instruct the user to add the label to the node before applying the file.

Reply:
I think adding what you have for the example is sufficient for the website.